packages/vscode-extension/src/devices/DeviceManager.ts

```diff
@@ -104,13 +105,20 @@ export class DeviceManager implements DeviceManagerInterface {
       Logger.error("Error fetching emulators", e);
       return [];
     });
-    const simulators =
-      Platform.OS === "macos"
-        ? listSimulators().catch((e) => {
-            Logger.error("Error fetching simulators", e);
-            return [];
-          })
-        : Promise.resolve([]);
+
+    let shouldLoadSimulators = Platform.OS === "macos";
+
+    if (!(await checkXcodeExists())) {
```

Comment:
do we even want to run those checks if we're not on a Mac? 

Perhaps doing it this way would be cleaner:
```ts
const shouldLoadSimulators = Platform.OS === "macos" && (await checkXcodeExists());
```

Then you can move the log to be printed in `checkXcodeExists`

Reply:
I did this in this way to address your previous comment about adding logging in !checkXcodeExist() scennario